src/core/src/pass/serialize.cpp

```diff
@@ -996,11 +997,35 @@ void serialize_rt_info(pugi::xml_node& root, const std::string& name, const ov::
             serialize_rt_info(child, it.first, it.second);
         }
     } else {
-        std::string value = data.as<std::string>();
+        const auto& value = data.as<std::string>();
         child.append_attribute("value").set_value(value.c_str());
     }
 }
 
+bool append_custom_rt_info(pugi::xml_node& node, const std::string& name, const ov::Any& data) {
```

Comment:
should custom RT info be included  hash calculations?
If some model has some custom RT info which was not serialized it will change the model hash.
If the custom RT info is not deterministic the cache feature may not work

Reply:
> Should custom RT info be included hash calculations?

It's safer not to include custom rt_info entries in hash calculations - PR changes accordingly.

Reply:
Such tests are already [there](https://github.com/t-jankowski/openvino/blob/43863a4cf37715dd50b040a71374462e1280fe98/src/inference/tests/unit/compilation_context_test.cpp#L252-L288). It's handled by [ModelCache::compute_hash](https://github.com/t-jankowski/openvino/blob/43863a4cf37715dd50b040a71374462e1280fe98/src/inference/src/dev/compilation_context.cpp#L71-L83).

Reply:
I extended custom info serialization tests `ov::pass::Hash' output value is unaffected [here](https://github.com/openvinotoolkit/openvino/pull/30784/commits/c98e042edf79fa901f543e33e03aecc143d64c16).